### ISSUE TITLE
No `needless_liftimes` in impl

### ISSUE THREAD

**Issue by jbaum98** [rust-lang/rust-clippy#4426]
Clippy doesn't report a `needless_lifetimes` lint in this code:

```rust
struct Struct {}

impl Struct {
   fn f<'a>(&'a self, x: &'a str) {
        println!("{}", x)
    } 
}
```

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=%23!%5Ballow(dead_code)%5D%0A%0Astruct%20Struct%20%7B%7D%0A%0Aimpl%20Struct%20%7B%0A%20%20%20fn%20f%3C'a%3E(%26'a%20self%2C%20x%3A%20%26'a%20str)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22%7B%7D%22%2C%20x)%0A%20%20%20%20%7D%20%0A%7D%0A)

If you remove the `self` parameter, `clippy` correctly reports the lint.

**Comment by mikerite** [comment]
This isn't a bug. `f(&self, x: &str)` is equivalent to `f<'a,'b>(&'a self, &'b str)` and not `f<'a>(&'a self, &'a str)`.

**Comment by jbaum98** [comment]
Oof I'm sorry! I clearly misunderstood the lifetime elision rules.
